saleor/plugins/webhook/subscription_webhooks/fixtures.py

```diff
@@ -0,0 +1,880 @@
+    )
+    webhook.events.create(event_type=WebhookEventAsyncType.PRODUCT_DELETED)
+    return webhook
+
+
+PRODUCT_VARIANT_CREATED_SUBSCRIPTION_QUERY = """
+    subscription{
+      event{
+        ...on ProductVariantCreated{
+          productVariant{
+            id
+          }
+        }
+      }
+    }
+"""
+
+
+@pytest.fixture
+def subscription_product_variant_created_webhook(app):
+    webhook = Webhook.objects.create(
+        name="Subscription product variant webhook",
+        app=app,
+        target_url="http://www.example.com/any",
+        subscription_query=PRODUCT_VARIANT_CREATED_SUBSCRIPTION_QUERY,
+    )
+    webhook.events.create(event_type=WebhookEventAsyncType.PRODUCT_VARIANT_CREATED)
+    return webhook
+
+
+PRODUCT_VARIANT_UPDATED_SUBSCRIPTION_QUERY = """
+    subscription{
+      event{
+        ...on ProductVariantUpdated{
+          productVariant{
+            id
+          }
+        }
+      }
+    }
+"""
+
+
+@pytest.fixture
+def subscription_product_variant_updated_webhook(app):
+    webhook = Webhook.objects.create(
+        name="Subscription product variant webhook",
+        app=app,
+        target_url="http://www.example.com/any",
+        subscription_query=PRODUCT_VARIANT_UPDATED_SUBSCRIPTION_QUERY,
+    )
+    webhook.events.create(event_type=WebhookEventAsyncType.PRODUCT_VARIANT_UPDATED)
+    return webhook
+
+
+PRODUCT_VARIANT_DELETED_SUBSCRIPTION_QUERY = """
+    subscription{
+      event{
+        ...on ProductVariantDeleted{
+          productVariant{
+            id
+          }
+        }
+      }
+    }
+"""
+
+
+@pytest.fixture
+def subscription_product_variant_deleted_webhook(app):
+    webhook = Webhook.objects.create(
+        name="Subscription product variant webhook",
+        app=app,
+        target_url="http://www.example.com/any",
+        subscription_query=PRODUCT_VARIANT_DELETED_SUBSCRIPTION_QUERY,
+    )
+    webhook.events.create(event_type=WebhookEventAsyncType.PRODUCT_VARIANT_DELETED)
+    return webhook
+
+
+PRODUCT_VARIANT_OUT_OF_STOCK_SUBSCRIPTION_QUERY = """
+    subscription{
+      event{
+        ...on ProductVariantOutOfStock{
+          productVariant{
+            id
+          }
+        }
+      }
+    }
+"""
+
+
+@pytest.fixture
+def subscription_product_variant_out_of_stock_webhook(app):
+    webhook = Webhook.objects.create(
+        name="Subscription product variant webhook",
+        app=app,
+        target_url="http://www.example.com/any",
+        subscription_query=PRODUCT_VARIANT_OUT_OF_STOCK_SUBSCRIPTION_QUERY,
+    )
+    webhook.events.create(event_type=WebhookEventAsyncType.PRODUCT_VARIANT_OUT_OF_STOCK)
+    return webhook
+
+
+PRODUCT_VARIANT_BACK_IN_STOCK_SUBSCRIPTION_QUERY = """
+    subscription{
+      event{
+        ...on ProductVariantBackInStock{
+          productVariant{
+            id
+          }
+        }
+      }
+    }
+"""
+
+
+@pytest.fixture
+def subscription_product_variant_back_in_stock_webhook(app):
```

Comment:
Looks like alls fixtures look very similar. You could do something like this:
```
@pytest.fixture
def subscription_webhook(app):
    def fun(query, event_type, name="Subscription"):
        webhook = Webhook.objects.create(
            name=name,
            app=app,
            target_url="http://www.example.com/any",
            subscription_query=query,
        )
        webhook.events.create(
            event_type=event_type
        )
        return webhook
@pytest.fixture
def subscription_product_variant_back_in_stock_webhook(subscription_webhook):
    return subscription_webhook(PRODUCT_VARIANT_BACK_IN_STOCK_SUBSCRIPTION_QUERY, WebhookEventAsyncType.PRODUCT_VARIANT_BACK_IN_STOCK)

``` 

This should be re-usable between all events without duplicate a lot of code.

Reply:
brilliant. Fixed it